### PR TITLE
Get a11y lint script paths from rails helper

### PR DESCRIPTION
**Pull request by aitchiss**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

The accessibility linting feature introduced in #13745 currently isn't working in non-Heroku Forems. This is because the JS is looking for an asset path which doesn't include the fingerprinted asset name, and it's therefore not being resolved or loaded in.

Rails helpers allow us to get the accurate asset path, and as these are needed in the Preact code, I've added them as data attributes on a hidden `div` that we can then retrieve from the Preact code. This is a similar pattern to what we do in various places where data needs passed from Rails to the Javascript code, and I've added a comment in the JS to explain why we're doing this.

## Related Tickets & Documents

#13745

## QA Instructions, Screenshots, Recordings

To test this, we need to check that the accessibility linter is functional in both the "Edit post" and "New post" views. Ideally we'd test this in an environment that relied on accessing fingerprinted versions of the assets, but I might need advice on how to do that!

https://user-images.githubusercontent.com/20773163/120770014-9b92b400-c515-11eb-8ffc-6a76f8c21dfc.mp4

### UI accessibility concerns?

N/A - this only changes how the script is loaded in

## Added tests?

- [ ] Yes
- [X] No, and this is why: the current tests test the feature
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [X] This change does not need to be communicated, and this is why not: we haven't communicated the original feature yet

## [optional] Are there any post deployment tasks we need to perform?

We may need to re-check this functionality in the fleet, to make sure it's definitely working

### CODE REVIEW

**Review by jacobherrington — APPROVED** [review]

File: app/views/articles/edit.html.erb [hunk 1]
```diff
@@ -31,3 +31,4 @@
 <%= render "articles/v2_form", article: @article, organizations: @organizations, version: @version %>
 
 <%= javascript_packs_with_chunks_tag "clipboardCopy", "articleForm", defer: true %>
+<div id="markdown-lint-js-path" class="hidden" data-markdown-it-js-path="<%= asset_path("markdown-it.min.js") %>" data-markdown-lint-js-path="<%= asset_path("markdownlint-browser.min.js") %>">
```
Comment (rhymes) [inline app/views/articles/edit.html.erb:34]: would it be possible to move these two files in a vendor dir, like `vendor/assets/javascripts`? See https://guides.rubyonrails.org/asset_pipeline.html#asset-organization

**Review by rhymes — COMMENTED** [review]
Left a note about where to put the files.

It's probably best to test this on a canary Forem but you'll need the help of the SRE team...

File: app/views/articles/edit.html.erb (hunk 1 above)
Comment (aitchiss) [reply app/views/articles/edit.html.erb:34]: Good point! I've done that now - I had to add some additional config to `assets.rb` to get them precompile, but locally it seems to be working now

File: config/initializers/assets.rb [hunk 2]
```diff
@@ -11,7 +11,10 @@
 
 Rails.application.config.assets.paths << Rails.root.join("app/assets/xml")
 
+Rails.application.config.assets.paths << Rails.root.join("vendor/assets/javascripts")
+
```
Comment (rhymes) [inline config/initializers/assets.rb:15]: These aren't needed, `vendor/assets/javascripts` is already part of the paths where Rails looks for:

From `main`:

```ruby
> Rails.application.config.assets.paths.include?(Rails.root.join("vendor/assets/javascripts").to_s)
=> true
```

```suggestion

```

File: config/initializers/assets.rb [hunk 3]
```diff
@@ -11,7 +11,10 @@
...
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
+Rails.application.config.assets.precompile += %w[markdown-it.min.js markdownlint-browser.min.js]
```
Comment (rhymes) [inline config/initializers/assets.rb:19]: 🔥

**Review by rhymes — APPROVED** [review]
Left a suggestion but other than that, let's give this a try on a test Forem!

File: jest.config.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ module.exports = {
       statements: 42,
       branches: 38,
       functions: 41,
-      lines: 43,
+      lines: 42,
```
Comment (aitchiss) [inline jest.config.js:31]: This code is all tested via Cypress, which doesn't currently count toward coverage

**Review by djuber — APPROVED** [review]
I agree, we should deploy to a canary first.

**Review by Ridhwana — APPROVED** [review]

**Comment by andygeorge** [comment]
Deployed to canary1
```
0c3b6b1ef6f7  quay.io/forem/forem:pr-13912             bundle exec sidek...  3 minutes ago  Up 3 minutes ago  0.0.0.0:80->80/tcp  forem-worker
ae0e9ba9a5f3  quay.io/forem/forem:pr-13912             bundle exec rails...  3 minutes ago  Up 3 minutes ago  0.0.0.0:80->80/tcp  forem-rails
```

**Review by andygeorge — APPROVED** [review]

**Comment by aitchiss** [comment]
Double checked in canary1, and verified fingerprinted asset paths are retrieved:

screenshot of dev tools elements panel with scripts added with correct paths

And feature works as expected:

https://user-images.githubusercontent.com/20773163/121491174-a1810d00-c9cd-11eb-91db-c7d8f3f80c7d.mp4

**Pull request merged**